riptide-capture/src/main/java/org/zalando/riptide/capture/DefaultCapture.java

```diff
@@ -0,0 +1,63 @@
+package org.zalando.riptide.capture;
+
+/*
+ * ⁣​
+ * Riptide: Capture
+ * ⁣⁣
+ * Copyright (C) 2015 - 2016 Zalando SE
+ * ⁣⁣
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ​⁣
+ */
+
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureAdapter;
+
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicReference;
+
+final class DefaultCapture<T> implements Capture<T> {
+
+    private final AtomicReference<T> reference = new AtomicReference<>();
+
+    @Override
+    public void accept(final T input) throws IOException {
+        reference.set(input);
+    }
+
+    @Override
+    public T retrieve() throws NoSuchElementException {
+        final T value = reference.get();
+
+        if (value == null) {
+            throw new NoSuchElementException("No value present");
+        }
+
+        return value;
+    }
+
+    @Override
+    public ListenableFuture<T> adapt(final ListenableFuture<?> future) {
+        @SuppressWarnings("unchecked") // TODO fix
```

Comment:
(1) and (2). And the only thing I was able to contribute on mobile for now (90 files changed seems to be the cap for my device :wink:)